src/loss_and_reg.jl

```diff
@@ -117,17 +141,32 @@ function avgerror(a::AbstractArray, l::ordinal_hinge)
     sum(map(ai->evaluate(l,m,ai),a))/length(a)
 end
 
-function avgerror(a::AbstractArray, l::hinge)
```

Comment:
hinge is subsumed by imbalanced_hinge, which has its own avgerror(). Something created with hinge() will be of type imbalanced_hinge with case_weight_ratio=1.